### ISSUE TITLE
Get download link by both choosing the quality and the extension

### ISSUE THREAD

**Issue by Knocks83** [ytdl-org/youtube-dl#28391]
## Checklist

- [X] I'm asking a question
- [X] I've looked through the README and FAQ for similar questions
- [X] I've searched the bugtracker for similar questions including closed ones

## Question

Hello, I'm trying to get some video download links using Youtube-DL using Python. I'm experiencing something like a problem: basically I need the best MP4 video available, but ydl gives me a m3u8 file. For now, I tried using best[ext=mp4] as format, but it didn't work as I expected (extract_info[url] it's still them m3u8 file). Do I have to loop through the files to find the best one myself or is there a way to say "I want the best one with that extension" or "I want the best one that DOESN'T HAVE this extension"?

**Comment by dstftw** [comment]
Read FAQ on format selection.

**Comment by Knocks83** [comment]
Ok cool, I can choose which extension I want to use but can I exclude an extension?

**Comment by Knocks83** [comment]
Another thing: I just tried using format: mp4 but it doesn't work in a website that isn't YouTube. To check if I was doing things correctly I tried getting the URL of a YouTube video with both mp4 (which gave me a video) and bestaudio (which gave me an audio).
When I try getting this video link (I'm trying to make a Telegram bot, the video is getting downloaded from XNXX) with format mp4 I still get a m3u8 file. The list of formats has an MP4 file.
